elastic/src/main/java/org/openmbee/mms/elastic/utils/BulkProcessor.java

```diff
@@ -0,0 +1,65 @@
+package org.openmbee.mms.elastic.utils;
+
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.update.UpdateRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.openmbee.mms.core.exceptions.InternalErrorException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+public class BulkProcessor {
+
+    protected int bulkLimit;
+    RestHighLevelClient client;
+    RequestOptions options;
+    List<DocWriteRequest<?>> allRequests = new ArrayList<>();
```

Comment:
this can just be BulkRequest, then bulkBatchRequests don't need to readd them all when sending